sympy/functions/combinatorial/numbers.py

```diff
@@ -1858,12 +1858,27 @@ def stirling(n, k, d=None, kind=2, signed=False):
 
 
 @cacheit
-def _nT(n, k):
+def _nT(n, k=None):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if k is None:
+        return partition(n)
+    if k > n:
+        return 0
+    if 2*k >= n:
+        return partition(n - k)
```

Comment:
I think @jksuom mentioned `_partition` so you will want to call this as `partition._partition(n - k)`